openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -309,8 +314,98 @@ def _upgrade_reminder_schedules_for_bin(site, target_day, bin_num, org_list, exc
         yield (user, first_schedule.enrollment.course.language, template_context)
 
 
+class CourseUpdate(ScheduleMessageType):
+    pass
+
+
+@task(ignore_result=True, routing_key=ROUTING_KEY)
+def course_update_schedule_bin(
+    site_id, target_day_str, day_offset, bin_num, org_list, exclude_orgs=False, override_recipient_email=None,
+):
+    target_day = deserialize(target_day_str)
+    msg_type = CourseUpdate()
+
+    for (user, language, context) in _course_update_schedules_for_bin(
+        Site.objects.get(id=site_id),
+        target_day,
+        day_offset,
+        bin_num,
+        org_list,
+        exclude_orgs
+    ):
+        msg = msg_type.personalize(
+            Recipient(
+                user.username,
+                override_recipient_email or user.email,
+            ),
+            language,
+            context,
+        )
+        _course_update_schedule_send.apply_async((site_id, str(msg)), retry=False)
```

Comment:
`str`? Any issues with this and non-ASCII messages?

Reply:
Yeah, we're in python2 land, so we should use `unicode` on the messages, rather than `str` like we do in the edx-ace repo.

Reply:
good catch.  updated.